packages/flame/lib/src/components/base_component.dart

```diff
@@ -154,6 +154,8 @@ abstract class BaseComponent extends Component {
     if (this is HasGameRef) {
       final c = this as HasGameRef;
       gameRef ??= c.hasGameRef ? c.gameRef : null;
+    } else if (gameRef == null && isMounted) {
+      throw 'Parent was already added to Game and has no HasGameRef; in this case, gameRef is mandatory.';
```

Comment:
How about making it Unsupported error? 

Using direct strings makes impossible to use try-on syntax.

Reply:
Or an assert maybe? This should never be try-catched on.

Reply:
ah yes this is never supposed to be caught, it's a "RuntimeException". shows a developer error, not a user error. is assert the idiomatic way? I guess that is similar to what we do for tappables, etc, right? I can replace

Reply:
changing to assert just like the other places!